pytomo3d/window/window.py

```diff
@@ -58,8 +59,81 @@ def load_window_config_yaml(filename):
     return pyflex.Config(**data)
 
 
+def update_user_levels(user_module, config, station, event, obsd, synt):
+    """Update user levels as an array using user_module
+
+    This function emulates the user_functions feature in FLEXWIN. User
+    has to prepare a python module that has a method called
+    generate_user_levels. This module has to be passed out as a string
+    (It should be formatted as you would in an import statement). And
+    then, user_module is dynamically imported and the
+    generate_user_levels function is called with the following
+    parameters: pyflex config, obspy station data, obspy event data,
+    observed and synthetic obspy traces. It is excepted that user to
+    use given scalar acceptance levels as base values and generate
+    acceptance level arrays as long as trace data to be given to the
+    pyflex. Function should return these values in order: stalta water
+    level, tshift acceptance level, dlna acceptance level,
+    cc_acceptance_level, s2n_limit.
+
+    :param user_module: user module as a string
+    :type user_module: str
+    :param config: window selection config
+    :type config_dict: pyflex.Config
+    :param station: station information which provids station location to
+        calculate the epicenter distance
+    :type station: obspy.Inventory or pyflex.Station
+    :param event: event information, providing the event information
+    :type event: pyflex.Event, obspy.Catalog or obspy.Event
+    :param observed: observed trace
+    :type observed: obspy.Trace
+    :param synthetic: synthetic trace
+    :type synthetic: obspy.Trace
+    :return: window selection config with arrays
+    :rtype: pyflex.Config
+    """
+    # Ridvan Orsvuran, 2016
+    # If user gives a user_module, allow it to create the user
+    # acceptance levels as arrays.
+    try:
+        # Add current working directory to path. This enables user to
+        # use python files in the working directory.
+        import sys
+        sys.path.append(".")
+
+        # Import the user module
+        user = importlib.import_module(user_module)
+        # Assign generate function to a variable. This enables to
```

Comment:
"user_module" should be in the python path. It could be in the same directory or not. It works like the import statement.
For example,  you can create a package called "my_user_functions" which contains a couple of user functions and install it using setup.py. Then you can set user_module to `my_user_module.majorarcwaves` or `my_user_module.myfrequencyrange_zcomp`.

Basically, if you can import the `user_module` in a python file, you can import using `importlib.import_module`.

> I think a easy solution would be just integrate this part into the code(rather than conditional import at run time) and provide several options. 

I am not really sure what you are suggesting here.